### ISSUE TITLE
torch.ops.image.decode_image(called from torchvision.io.read_image) crashes python on this image

### ISSUE THREAD

**Issue by you74674** [pytorch/vision#4485]
### 🐛 Describe the bug

```torchvision.io.read_image``` crashes python when reading this image(from 300W/lfpw/trainset dataset).
There's no error message, but it looks like the crash happens in ```torch.ops.image.decode_image```.
I have tested it on three different computer(0.10.0, 0.10.1 and 0.10.1+cpu), and all of them crashed.

```python
from PIL import Image
import numpy as np
img = np.array(Image.open('image_0779.png'))#this is fine
import torchvision
torchvision.io.read_image('image_0779.png')#this crash
```

![image_0779](https://user-images.githubusercontent.com/14540041/134813178-bdea78c3-aebb-4db1-9b87-5efd73fbc99c.png)

### Versions

Collecting environment information...
PyTorch version: 1.9.1+cu102
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 10 Home Insider Preview
GCC version: Could not collect
Clang version: Could not collect
CMake version: version 3.21.2
Libc version: N/A

Python version: 3.7.6 (tags/v3.7.6:43364a7ae0, Dec 19 2019, 00:42:30) [MSC v.1916 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-10-10.0.21296-SP0
Is CUDA available: True
CUDA runtime version: 11.1.74

GPU models and configuration: GPU 0: NVIDIA TITAN Xp COLLECTORS EDITION
Nvidia driver version: 510.10
cuDNN version: C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.2\bin\cudnn_ops_train64_8.dll
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] Could not collect
[conda] Could not collect

**Comment by prabhat00155** [comment]
On Mac, this throws the following error:
```
RuntimeError: At most 8-bit PNG images are supported currently.
```
Your image has a bit-depth of 16, which is [currently not supported](https://github.com/pytorch/vision/blob/b7c235892332c49a86c4be1b00fb6a672a3f9930/torchvision/csrc/io/image/cpu/decode_png.cpp#L75).
That said, it should not crash your Python. I'll try to reproduce this on Windows.

**Comment by prabhat00155** [comment]
Thanks for reporting the bug. I was able to reproduce this issue with 0.10 on Windows. However, I think this issue is same as https://github.com/pytorch/vision/issues/3613 which was fixed in https://github.com/pytorch/vision/pull/4101. If you try the latest nightly(`conda install pytorch torchvision torchaudio cpuonly -c pytorch-nightly`), you wouldn't see the crash with the correct exception thrown.
